### PR TITLE
Laura Lyckholm Project Mongo API

### PR DESCRIPTION
**Pull request by LauraLyckholm**
https://netflix-mongo-api.onrender.com/

### CODE REVIEW

File: routes/movieRoute.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        })
+        .catch(error => {
+            res.status(500).json({ error: "Something went wrong, please try again." });
+        });
+});
+
+// Gets an individual movie based on its id. 
+router.get("/movies/:id", (req, res) => {
+    const movieID = req.params.id; // Gets the id from the params
+    MovieModel.findOne({ _id: movieID }).then(movie => { // Checks if the id from the param is the same as the show_id, if it then it's displayed in json. Findone, because I only want one result.
```
Comment (AnnaRobertsson) [inline routes/movieRoute.js:51]: Did you consider using findById instead? Both works fine though! :)

File: routes/movieRoute.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        })
+        .catch(error => {
+            res.status(500).json({ error: "Something went wrong, please try again. If your having problems with the query, check the readme for further instructions." });
+        });
+});
+
+// Makes it possible to query on a string included in the title
+router.get("/movies/title/:title", (req, res) => {
+    const titleQuery = req.params.title.toLowerCase();
+    MovieModel.find({ title: { $regex: titleQuery, $options: "i" } }) // Query-filter. Search is performed on "title", titleQuery is the regex pattern searched for, options: "i" makes the search case-insensitive
```
Comment (AnnaRobertsson) [inline routes/movieRoute.js:89]: I think it's a nice feature that you can type in part of the title and get options that contain the word, for example typing in "rain" and get titles like "Rain man", "Trainspotting" and "The Rainmaker" etc. Good job!

**Review by AnnaRobertsson — COMMENTED** [review]
Great job Laura with this assignment! All your endpoints work beautifully! And you have really made an excellent job with the error handling throughout the project! As always, your code looks really neat and arranged with descriptive comments to guide the one looking through the code :) Be proud!

File: models/MovieModel.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    date_added: { type: String },
+    release_year: { type: Number },
+    rating: { type: String },
+    duration: { type: String },
+    listed_in: { type: String },
+    description: { type: String },
+    type: { type: String }
+});
+
+export const MovieModel = mongoose.model("Movie", movieSchema);
```
[the remark this answers is not in the document]
Comment (LauraLyckholm) [reply models/MovieModel.js:20]: That's actually a really good point... type: [String] is probably referring to those parts being arrays of string.. and There MIGHT be more than one castmember etc yes :D  I should look this up!

File: data/netflix-titles.json [hunk 4]
```diff
@@ -19249,4 +19249,4 @@
     "description": "Undercover agents infiltrate a drug kingpin's operation by posing as a couple at the campground where he spends his weekends. Inspired by real events.",
     "type": "TV Show"
   }
-]
+]
```
[the remark this answers is not in the document]
Comment (LauraLyckholm) [reply data/netflix-titles.json:19252]: I think I might just have removed an empty row and then added it again, to "not make changes" to the data haha :D

File: models/MovieModel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnnaRobertsson) [reply models/MovieModel.js:20]: I remember I also saw some movies in the database with multiple countries listed so maybe add in that as well.. :)

**Pull request closed without merging**